src/components/navigation/navigation.tsx

```diff
@@ -152,62 +146,37 @@ export class CalciteNavigation {
     );
   }
 
-  renderNavLevel(level: Level): VNode {
-    const hasElements = this.hasSlottedElements(level);
-    const isPrimaryLevel = level === LEVEL.primary;
-    return (
-      <div
-        class={{
-          [CSS.container]: true,
-          [level]: true,
-          hide: !hasElements
-        }}
-      >
-        <slot name={SLOTS.progress} />
-        <div class={CSS.containerContent}>
-          {isPrimaryLevel && this.renderMenuAction()}
-          {isPrimaryLevel && <slot name={SLOTS.logo} onSlotchange={this.handleLogoSlotChange} />}
-          <slot
-            data-level={level}
-            name={
-              isPrimaryLevel
-                ? SLOTS.primaryContentStart
-                : level === LEVEL.secondary
-                ? SLOTS.secondaryContentStart
-                : SLOTS.tertiaryContentStart
-            }
-            onSlotchange={this.handleContentSlotChange}
-          />
-          {isPrimaryLevel && (
-            <slot
-              data-level={level}
-              name={SLOTS.primaryContentCenter}
-              onSlotchange={this.handleContentSlotChange}
-            />
-          )}
-          <slot
-            data-level={level}
-            name={
-              isPrimaryLevel
-                ? SLOTS.primaryContentEnd
-                : level === LEVEL.secondary
-                ? SLOTS.secondaryContentEnd
-                : SLOTS.tertiaryContentEnd
-            }
-            onSlotchange={this.handleContentSlotChange}
-          />
-          {isPrimaryLevel && <slot name={SLOTS.user} onSlotchange={this.handleUserSlotChange} />}
-        </div>
-      </div>
-    );
-  }
-
   render(): VNode {
+    const isPrimaryLevel =
+      this.el.slot !== SLOTS.navSecondary && this.el.slot !== SLOTS.navTertiary;
     return (
       <Host>
-        {this.renderNavLevel("primary")}
-        {this.renderNavLevel("secondary")}
-        {this.renderNavLevel("tertiary")}
+        <div
+          class={{
+            [CSS.container]: true,
+            [SLOTS.navSecondary]: this.el.slot === SLOTS.navSecondary,
+            [SLOTS.navTertiary]: this.el.slot === SLOTS.navTertiary
+          }}
+        >
+          {isPrimaryLevel && <slot name={SLOTS.progress} />}
```

Comment:
With all this `isPrimaryLevel` logic that doens't apply to secondary/tertiary, I'm thinking maybe "primary" is a separate component that uses the "secondary/tertiary" component inside it? 

Only the "primary" component should have the slots for "secondary" and "tertiary".

It would also clean up all this logic depending on the slot name.

Something like:

```html
<calcite-navigation>
  <calcite-subnavigation slot="secondary"></calcite-subnavigation>
  <calcite-subnavigation slot="tertiary"></calcite-subnavigation>
</calcite-navigation>
```

Internally to "calcite-navigation" it could use the "calcite-subnavigation" component to reduce rendering logic as well.

Reply:
I'd prefer to avoid a second component, as I think that complicates the story quite a bit - this is already pretty simplified compared to the original attempt with a single component and all the slots. Open to better ways to manage the slotted position though, maybe an internal prop / state vs. checking in render.

Reply:
Yeah I talked to @anveshmekala about a `@State` which makes sense. 

Something like...

```tsx
@State() activeLevel: "secondary" | "primary" | "tertiary";

// onslotchange do this
this.activeLevel = "secondary"// or whatever it is based on this.el.slot

// Then in render you can just use
const isPrimaryLevel = this.activeLevel === "primary";
```

Reply:
The only weird part is that some slots would be conditional.

Instead of conditionally rendering the slots, we should hide the slots. Otherwise, the `onslotchange` event won't ever be fired if the slot isn't first rendered.